### PR TITLE
remove previously deprecated code

### PR DESCRIPTION
**Pull request by alanhamlett**
To get ready for a new feature release as per [this comment](https://github.com/maxcountryman/flask-login/discussions/670#discussioncomment-2768626).

* `header_loader`
* `user_loaded_from_header`
* `setup_app`
* `_login_disabled`

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -543,8 +515,6 @@ Configuring Login
 
 .. autoclass:: LoginManager
 
-   .. automethod:: setup_app
```
Comment (tiesjan) [inline docs/index.rst:546]: Perhaps this can be replaced with `init_app` instead, such that initialization is still documented.

**Review by tiesjan — APPROVED** [review]
Looks good! One minor suggestion below. 👍🏻

**Review by davidism — APPROVED** [review]

**Comment by davidism** [comment]
For now I'm calling this 0.7 instead of 1.0.

**Comment by maxcountryman** [comment]
> For now I'm calling this 0.7 instead of 1.0.

I'm in favor of sticking to 0.x releases. While I don't expect flask-login to change much, I don't feel confident there aren't still things that should be thought through before declaring it 1.0.

**Pull request merged**